cmake/JuliaCCompiler.cmake

```diff
@@ -0,0 +1,84 @@
+#
+
+# libc++ is standard on OS X 10.9, but not for earlier releases
+jl_option(USE_LIBCPP "Use libc++" Off)
+# assume we don't have LIBSSP support in our compiler,
+# will enable later if likely true
+set(HAVE_SSP Off)
+
+jl_set_option(USEICC Off)
+jl_set_option(USEGCC Off)
+jl_set_option(USECLANG Off)
+jl_set_option(USEMSVC Off)
```

Comment:
> You need to check CMAKE_SIZEOF_VOID_P to distinguish 32 vs 64 bit.

Good to know this is the right approach. I think I use that somewhere already.

> We should probably error out for CYGWIN with the suggestion to cross-compile, and put in a toolchain file for that eventually. Libuv can't build with cygwin or msys host, though I think the cmake MSYS variable is about the build environment rather than the compiler host.

I see.

Would you mind putting these todo's to a comment in the PR in order not to lose them on a update? Thx.